### PR TITLE
Disable use of min/max macros defined by Windows headers.

### LINKED ISSUE #362: Disable Windows-provided min/max macros
`minwindef.h` defines macros for min/max functionality.  For places of the codebase where min/max is required, the code defaults to using these macros instead of the `std::min`/`std::max`.  The macros can be disabled with the `NOMINMAX` flag.  We should disable the macros in-order to improve the portability of our code.  Related to #212.

### PR DESCRIPTION
**Pull request by danbelcher-MSFT**
### Description of the changes:
- Disable Windows-provided min/max macros using the `NOMINMAX` flag.  Add the flag to each project's pch to disable the macros across the solution.

### How changes were validated:
- Project builds.
- Unit tests pass.
- Smoke tests.

### CODE REVIEW

**Comment by rudyhuyn** [comment]
This is great! I faced this issue some days ago when I was testing some Lexers/Parsers and were forced to manually modify source codes generated by Yacc/Bison. Good job!

File: src/Calculator/Controls/CalculationResult.cpp [hunk 1]
```diff
@@ -207,10 +207,10 @@ void CalculationResult::UpdateTextState()
     {
         double widthDiff = abs(m_textBlock->ActualWidth - containerSize);
         double fontSizeChange = INCREMENTOFFSET;
-
+
```
Comment (janisozaur) [inline src/Calculator/Controls/CalculationResult.cpp]: Whitespace

**Comment by janisozaur** [comment]
Nice. If possible, I'd like this to be merged first and I will rebase #212 and fix the issues I ran into in that PR

File: src/Calculator/Controls/CalculationResult.cpp (hunk 1 above)
Comment (danbelcher-MSFT) [reply src/Calculator/Controls/CalculationResult.cpp]: Hmmm

File: src/CalcManager/Ratpack/logic.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@
 #include "pch.h"
 #include "ratpak.h"
 
-
+using namespace std;
```
Comment (dertuxmalwieder) [inline src/CalcManager/Ratpack/logic.cpp]: Honestly, this is awful IMO.

Comment (danbelcher-MSFT) [reply src/CalcManager/Ratpack/logic.cpp]: Some people like it and some don't.  The simple answer here is that this is consistent with the rest of the codebase.  If the community wants to have a discussion about changing this precedent, someone can make an issue and we can have an open conversation.

**Pull request merged**